ingestion/src/metadata/profiler/adaptors/mongodb.py

```diff
@@ -11,10 +11,72 @@
 """
 MongoDB adaptor for the NoSQL profiler.
 """
+import json
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import Dict, List, Optional, Union
+
 from pymongo import MongoClient
+from pymongo.command_cursor import CommandCursor
+from pymongo.cursor import Cursor
 
-from metadata.generated.schema.entity.data.table import Table
+from metadata.generated.schema.entity.data.table import Column, Table
 from metadata.profiler.adaptors.nosql_adaptor import NoSQLAdaptor
+from metadata.utils.sqa_like_column import SQALikeColumn
+
+
+class AggregationFunction(Enum):
+    SUM = "$sum"
+    MEAN = "$avg"
+    COUNT = "$count"
+    MAX = "$max"
+    MIN = "$min"
+
+
+class Executable:
+    def to_executable(self, client: MongoClient) -> Union[CommandCursor, Cursor]:
+        raise NotImplementedError
+
+
+@dataclass
```

Comment:
any reason to have this as dataclass? Not against it, but would be cool to centralize our modelling around pydantic if it does not matter here

Reply:
switched to pydantic models